bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorImpl.java

```diff
@@ -86,7 +83,7 @@ protected void deactivate() {
      * @throws NullPointerException if given config description URI or configuration parameters are null
      */
     @Override
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings({ "unchecked", "null" })
     public void validate(Map<String, Object> configurationParameters, URI configDescriptionURI) {
         Objects.requireNonNull(configurationParameters, "Configuration parameters must not be null");
```

Comment:
If the class is annotated `@NonNullByDefault` both parameters are `@NonNull`. Isn't this check unnecessary?

Reply:
Yes, and there are plenty more of them which can be omitted in the related validation classes. But let us focus on annotations and c-tor injection here and move other improvements into a follow-up PR.

Reply:
Fine with me. I don't understand why the build failed.